src/app/clusters/level-control/level-control.cpp

```diff
@@ -123,6 +123,137 @@ static bool shouldExecuteIfOff(EndpointId endpoint, CommandId commandId,
+            pairs[attributeCount].attributeValue = frequency;
+            attributeCount++;
+        }
+
+        app::DataModel::List<AttributeValuePair> attributeValueList(pairs, attributeCount);
+
+        return EncodeAttributeValueList(attributeValueList, serializedBytes);
+    }
+
+    /// @brief Default EFS interaction when applying scene to the OnOff Cluster
+    /// @param endpoint target endpoint
+    /// @param cluster  target cluster
+    /// @param serializedBytes Data from nvm
+    /// @param timeMs transition time in ms
+    /// @return CHIP_NO_ERROR if value as expected, CHIP_ERROR_INVALID_ARGUMENT otherwise
+    CHIP_ERROR ApplyScene(EndpointId endpoint, ClusterId cluster, const ByteSpan & serializedBytes,
+                          scenes::TransitionTimeMs timeMs) override
+    {
+        app::DataModel::DecodableList<Scenes::Structs::AttributeValuePair::DecodableType> attributeValueList;
+
+        ReturnErrorOnFailure(DecodeAttributeValueList(serializedBytes, attributeValueList));
+
+        size_t attributeCount = 0;
+        ReturnErrorOnFailure(attributeValueList.ComputeSize(&attributeCount));
+        VerifyOrReturnError(attributeCount <= kLevelMaxScenableAttributes, CHIP_ERROR_BUFFER_TOO_SMALL);
+
+        auto pair_iterator = attributeValueList.begin();
+
+        // The level control cluster should have a maximum of 2 attributes
+        uint8_t level = 0;
+        // TODO : Uncomment when frequency is supported by the level control cluster
+        // uint16_t frequency;
+        while (pair_iterator.Next())
+        {
+            auto & decodePair = pair_iterator.GetValue();
+
+            // If attribute ID was encoded, checks which attribute from LC cluster is there
+            switch (decodePair.attributeID)
+            {
+            case Attributes::CurrentLevel::Id:
+                level = static_cast<uint8_t>(decodePair.attributeValue);
+                break;
+            case Attributes::CurrentFrequency::Id:
+                // TODO : Uncomment when frequency is supported by the level control cluster
+                // frequency = static_cast<uint16_t>(decodePair.attributeValue);
+                break;
+            default:
+                return CHIP_ERROR_INVALID_ARGUMENT;
+            }
+        }
+        ReturnErrorOnFailure(pair_iterator.GetStatus());
+
+        // TODO : Implement action on frequency when frequency not provisional anymore
+        // if(LevelControlHasFeature(endpoint, LevelControl::Feature::kFrequency)){}
+        Status status;
+        CommandId command = LevelControlHasFeature(endpoint, LevelControl::Feature::kOnOff) ? Commands::MoveToLevelWithOnOff::Id
+                                                                                            : Commands::MoveToLevel::Id;
+
+        status = moveToLevelHandler(
+            endpoint, command, level, app::DataModel::MakeNullable<uint16_t>(static_cast<uint16_t>(timeMs / 100)),
```

Comment:
But if the level was null when we stored the scene... won't this just ignore the moveToLevelHandler call, at least when MaxLevel is 254?  This looks broken.